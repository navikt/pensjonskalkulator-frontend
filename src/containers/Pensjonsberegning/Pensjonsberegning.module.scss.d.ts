import globalClassNames from '../../style.d'
declare const classNames: typeof globalClassNames & {
  readonly loader: 'loader'
}
export = classNames
