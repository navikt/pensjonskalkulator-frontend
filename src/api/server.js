// const express = require('express')
// const liveness = require('../src/__mocks__/data/liveness.json')
// const pensjonsberegning = require('../src/__mocks__/data/pensjonsberegning.json')

// const app = express()
// const port = 8088

// app.get('/api/status', (request, response) => {
//   response.send(liveness)
// })

// app.get('/api/pensjonsberegning', async (request, response) => {
//   response.send(pensjonsberegning)
// })

// app.listen(process.env.PORT || port, () => {
//   console.info('proxy server is running on port', process.env.PORT || port)
// })

// import { rest } from 'msw'
// import { setupServer } from 'msw/node'

import { setupServer } from 'msw/node'
import { handlers } from './handlers.js'

export const server = setupServer(...handlers)
server.listen({ onUnhandledRequest: 'error' })
