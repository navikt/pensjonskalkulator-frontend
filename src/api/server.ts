import { setupServer } from 'msw/node'
import { getHandlers } from './handlers.js'
import { rest } from 'msw'

export const API_TARGET = 'http://localhost:8088'
export const API_PATH = '/pensjon/kalkulator/api'

const handlers = getHandlers(`${API_TARGET}${API_PATH}`)
export const server = setupServer(...handlers)

server.listen({ onUnhandledRequest: 'error' })

type MockResponseOptions = {
  status?: number
  json?: any
}

export const mockResponse = (
  path: string,
  options: MockResponseOptions = {
    status: 200,
    json: 'OK',
  }
) => {
  server.use(
    rest.get(`${API_TARGET}${API_PATH}${path}`, (req, res, ctx) => {
      return res(ctx.status(options.status ?? 200), ctx.json(options.json))
    })
  )
}

export const mockErrorResponse = (
  path: string,
  options: MockResponseOptions = {
    status: 500,
    json: "Beep boop I'm an error!",
  }
) => {
  mockResponse(path, {
    ...options,
    status: options.status ?? 500,
  })
}
