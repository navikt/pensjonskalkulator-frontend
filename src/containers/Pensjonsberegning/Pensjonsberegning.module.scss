.container {
  composes: frame from '../../scss/modules/frame.module.scss';

  &__hasPadding {
    composes: frame__hasPadding from '../../scss/modules/frame.module.scss';
  }
}

.background {
  width: 100%;
  left: 0;
  right: 0;

  &__hasMargin {
    margin-top: var(--a-spacing-4);
    min-height: 33vh;
  }

  &__white {
    background-color: var(--a-white);
    composes: fadein from '../../scss/modules/fadein.module.scss';
  }

  &__lightblue {
    background-color: var(--a-deepblue-50);
  }
}
