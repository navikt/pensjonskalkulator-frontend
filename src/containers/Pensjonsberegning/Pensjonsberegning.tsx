import React, { useState } from 'react'

import { Alert, Heading } from '@navikt/ds-react'

import { Card } from '@/components/Card'
import { Forbehold } from '@/components/Forbehold'
import { Grunnlag } from '@/components/Grunnlag'
import { Loader } from '@/components/Loader'
import { Pensjonssimulering } from '@/components/Pensjonssimulering'
import { TidligstMuligUttaksalder } from '@/components/TidligstMuligUttaksalder'
import { TilbakeEllerAvslutt } from '@/components/TilbakeEllerAvslutt'
import { VelgUttaksalder } from '@/components/VelgUttaksalder'
import { useTidligsteUttaksalder } from '@/routes/hooks'

export function Pensjonsberegning() {
  const {
    data: tidligstMuligUttak,
    isLoading,
    isError,
    isSuccess,
  } = useTidligsteUttaksalder()

  const [valgtUttaksalder, setValgtUttaksalder] = useState<string | undefined>()

  if (isLoading) {
    return (
      <Loader
        data-testid="loader"
        size="3xlarge"
        title="Henter tidligste mulige uttaksalder"
      />
    )
  }

  if (isError || !isSuccess) {
    return (
      <Alert variant="error">
        <Heading spacing size="small" level="2">
          Vi klarte ikke å hente din tidligste mulige uttaksalder. Prøv igjen
          senere.
        </Heading>
      </Alert>
    )
  }

  return (
    <>
      <TidligstMuligUttaksalder uttaksalder={tidligstMuligUttak} />

      <Card>
        <VelgUttaksalder
          tidligstMuligUttak={tidligstMuligUttak}
          valgtUttaksalder={valgtUttaksalder}
          setValgtUttaksalder={setValgtUttaksalder}
        />

        {valgtUttaksalder && (
          <Pensjonssimulering uttaksalder={parseInt(valgtUttaksalder, 10)} />
        )}
      </Card>

      {valgtUttaksalder && (
        <>
          <Grunnlag tidligstMuligUttak={tidligstMuligUttak} />
          <Forbehold />
        </>
      )}

      <TilbakeEllerAvslutt />
    </>
  )
}
