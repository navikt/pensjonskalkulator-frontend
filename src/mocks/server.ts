import { rest } from 'msw'
import { setupServer } from 'msw/node'

import { API_BASEURL } from '@/paths'

import { getHandlers } from './handlers'

const handlers = getHandlers(API_BASEURL)
export const server = setupServer(...handlers)

type MockResponseOptions = {
  status?: number
  json?: ReturnType<typeof JSON.parse>
  text?: string
  method?: 'post' | 'get'
  baseUrl?: string
}

const defaultResponseOptions: Required<MockResponseOptions> = {
  status: 200,
  json: { data: 'OK' },
  method: 'get',
  baseUrl: `${API_BASEURL}`,
  text: '',
}

export const mockResponse = (
  path: string,
  inputOptions: MockResponseOptions = {}
) => {
  const options = { ...defaultResponseOptions, ...inputOptions }
  const parsedPath = `${options.baseUrl}${path}`

  server.use(
    rest[options.method](parsedPath, (_req, res, ctx) => {
      return res(
        ctx.status(options.status),
        options.text ? ctx.text(options.text) : ctx.json(options.json)
      )
    })
  )
}

const defaultErrorResponseOptions: Required<MockResponseOptions> = {
  status: 500,
  json: { data: "Beep boop I'm an error!" },
  text: '',
  method: 'get',
  baseUrl: `${API_BASEURL}`,
}

export const mockErrorResponse = (
  path: string,
  inputOptions: MockResponseOptions = {}
) => {
  const options = {
    ...defaultErrorResponseOptions,
    ...inputOptions,
  }
  mockResponse(path, options)
}
