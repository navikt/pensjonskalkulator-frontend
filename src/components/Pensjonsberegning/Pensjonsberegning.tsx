import React, { useMemo, useState } from 'react'

import { ChevronDownIcon, ChevronUpIcon } from '@navikt/aksel-icons'
import {
  Alert,
  Button,
  Chips,
  Heading,
  Ingress,
  Loader,
  ReadMore,
} from '@navikt/ds-react'
import clsx from 'clsx'

import { Pensjonssimulering } from '../Pensjonssimulering'
import { Card } from '@/components/Card'
import { Forbehold } from '@/components/Forbehold'
import { Grunnlag } from '@/components/Grunnlag'
import { useGetTidligsteMuligeUttaksalderQuery } from '@/state/api/apiSlice'

import { useAlderChips } from './hooks'
import { formatUttaksalder } from './utils'

import styles from './Pensjonsberegning.module.scss'

export function Pensjonsberegning() {
  const {
    data: tidligstMuligUttak,
    isLoading,
    isError,
    isSuccess,
  } = useGetTidligsteMuligeUttaksalderQuery()
  const DEFAULT_ANTALL_VISIBLE_ALDERCHIPS = 9
  const VIS_FLERE__ALDERE_LABEL_CLOSE = 'Vis flere aldere'
  const VIS_FLERE__ALDERE_LABEL_OPEN = 'Vis færre aldere'

  const [isFlereAldereOpen, setIsFlereAldereOpen] = useState<boolean>(false)

  const [valgtUttaksalder, setValgtUttaksalder] = useState<string | undefined>()

  const alderChips = useAlderChips(tidligstMuligUttak)
  const formatertUttaksalder = useMemo(() => {
    return isSuccess ? formatUttaksalder(tidligstMuligUttak) : null
  }, [isSuccess, tidligstMuligUttak])

  if (isLoading) {
    return (
      <Loader
        className={styles.loader}
        data-testid="loader"
        size="3xlarge"
        title="Henter tidligste mulige uttaksalder"
      />
    )
  }

  if (isError || !isSuccess) {
    return (
      <Alert variant="error">
        <Heading spacing size="small" level="2">
          Vi klarte ikke å hente din tidligste mulige uttaksalder. Prøv igjen
          senere.
        </Heading>
      </Alert>
    )
  }

  return (
    <>
      <Card data-testid="tidligst-mulig-uttak">
        <Ingress>
          Du kan tidligst ta ut alderspensjon når du er {formatertUttaksalder}.
          Hvis du går av senere, får du høyere pensjon i året.
        </Ingress>
      </Card>

      <Card className={styles.section}>
        <Heading size="xsmall" level="2">
          Når vil du ta ut alderspensjon?
        </Heading>
        <Chips
          className={clsx(styles.chipsWrapper, styles.chipsWrapper__hasGap)}
        >
          {alderChips.length > 0 &&
            alderChips
              .slice(
                0,
                isFlereAldereOpen
                  ? alderChips.length
                  : DEFAULT_ANTALL_VISIBLE_ALDERCHIPS
              )
              .map((alderChip) => (
                <Chips.Toggle
                  selected={valgtUttaksalder === alderChip}
                  key={alderChip}
                  onClick={() => setValgtUttaksalder(alderChip)}
                >
                  {alderChip}
                </Chips.Toggle>
              ))}
        </Chips>
        <Button
          className={styles.visFlereAldere}
          icon={
            isFlereAldereOpen ? (
              <ChevronUpIcon aria-hidden />
            ) : (
              <ChevronDownIcon aria-hidden />
            )
          }
          iconPosition="left"
          size={'xsmall'}
          variant="tertiary"
          onClick={() => {
            setIsFlereAldereOpen(!isFlereAldereOpen)
          }}
        >
          {isFlereAldereOpen
            ? VIS_FLERE__ALDERE_LABEL_OPEN
            : VIS_FLERE__ALDERE_LABEL_CLOSE}
        </Button>

        {valgtUttaksalder && (
          <>
            <Pensjonssimulering uttaksalder={parseInt(valgtUttaksalder, 10)} />
            <ReadMore
              header={'Vis tabell'}
              className={styles.visTabell}
              // icon={<ChevronDownIcon aria-hidden />}
              // iconPosition="left"
              // size={'medium'}
              // variant="tertiary"
            >
              <p>{'//TODO'}</p>
            </ReadMore>
          </>
        )}
      </Card>

      {valgtUttaksalder && (
        <>
          <Grunnlag tidligstMuligUttak={tidligstMuligUttak} />
          <Forbehold />
        </>
      )}
    </>
  )
}
