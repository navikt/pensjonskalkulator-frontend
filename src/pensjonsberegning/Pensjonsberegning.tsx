import { useEffect, useState } from 'react'

import { Alert, BodyLong, Loader } from '@navikt/ds-react'

import { fetchPensjonsberegning } from '@/api/pensjonsberegning'

import styles from './Pensjonsberegning.module.scss'

const usePensjonsberegning = (): FetchedData<PensjonsberegningResponse[]> => {
  const [beregning, setBeregning] = useState<
    FetchedData<PensjonsberegningResponse[]>
  >({
    data: null,
    isLoading: true,
    hasError: false,
  })

  useEffect(() => {
    fetchPensjonsberegning()
      .then((data: PensjonsberegningResponse[]) => {
        setBeregning({
          data: data.sort((a, b) => a.pensjonsaar - b.pensjonsaar),
          isLoading: false,
          hasError: false,
        })
      })
      .catch((error) => {
        // TODO add error loggingto a server?
        console.warn(error)
        setBeregning({
          data: null,
          isLoading: false,
          hasError: true,
        })
      })
  }, [])

  return beregning
}

export function Pensjonsberegning() {
  const beregning = usePensjonsberegning()

  if (beregning.isLoading) {
    return <Loader data-testid="loader" />
  } else if (beregning.hasError || beregning.data.length === 0) {
    return (
      <Alert variant="error">
        Vi klarte ikke å kalkulere pensjonen din. Prøv igjen senere.
      </Alert>
    )
  }

  return (
    <section className={styles.sammenligning}>
      <BodyLong>
        Hvis du fortsetter å ha samme inntekt som du har i dag kan du tidligst
        gå av med pensjon ved <b>{beregning.data[0].alder} år</b> hvor du vil få
        utbetalt <b>{beregning.data[0].pensjonsbeloep} kroner</b> i året
      </BodyLong>
      {beregning.data[1] && (
        <BodyLong>
          Dersom du jobber frem til du er <b>{beregning.data[1].alder} år</b>,
          vil du få <b>{beregning.data[1].pensjonsbeloep} kroner</b> utbetalt
        </BodyLong>
      )}
      {beregning.data[2] && (
        <BodyLong>
          Dersom du jobber frem til du er <b>{beregning.data[2].alder} år</b>,
          vil du få <b>{beregning.data[2].pensjonsbeloep} kroner</b> utbetalt
        </BodyLong>
      )}
    </section>
  )
}
