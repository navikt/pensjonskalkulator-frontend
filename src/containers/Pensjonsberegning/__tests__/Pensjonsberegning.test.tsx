import { describe, expect, it, vi } from 'vitest'

import { Pensjonsberegning } from '../Pensjonsberegning'
import { mockErrorResponse, mockResponse } from '@/mocks/server'
import * as apiSliceUtils from '@/state/api/apiSlice'
import { RootState } from '@/state/store'
import {
  render,
  screen,
  swallowErrorsAsync,
  userEvent,
  waitFor,
} from '@/test-utils'

describe('Pensjonsberegning', () => {
  it('har riktig sidetittel', () => {
    render(<Pensjonsberegning />)
    expect(document.title).toBe('application.title.beregning')
  })

  it('viser loading og deretter riktig header, tekst og knapper', async () => {
    const result = render(<Pensjonsberegning />)
    expect(screen.getByTestId('loader')).toBeVisible()
    await waitFor(() => {
      expect(screen.queryByTestId('loader')).not.toBeInTheDocument()
    })
    expect(screen.getByTestId('tidligst-mulig-uttak')).toBeVisible()
    expect(screen.getAllByRole('heading', { level: 2 })).toHaveLength(1)
    expect(screen.getAllByRole('button')).toHaveLength(12)
    expect(result.asFragment()).toMatchSnapshot()
  })

  it('viser feilmelding om henting av pensjonberegning feiler', async () => {
    mockErrorResponse('/tidligste-uttaksalder', {
      status: 500,
      json: "Beep boop I'm an error!",
      method: 'post',
    })

    const result = render(<Pensjonsberegning />)

    await waitFor(() => {
      expect(
        screen.getByText(
          'Vi klarte ikke å hente din tidligste mulige uttaksalder. Prøv igjen senere.'
        )
      ).toBeVisible()
      expect(result.asFragment()).toMatchSnapshot()
    })
  })

  it('viser feilmelding om pensjonsberegning er på ugyldig format', async () => {
    const invalidData = {
      aar: 67,
      maaned: null,
    } as unknown as Uttaksalder
    mockResponse('/tidligste-uttaksalder', {
      json: [invalidData],
      method: 'post',
    })

    render(<Pensjonsberegning />)

    await swallowErrorsAsync(async () => {
      await waitFor(() => {
        expect(
          screen.getByText(
            'Vi klarte ikke å hente din tidligste mulige uttaksalder. Prøv igjen senere.'
          )
        ).toBeVisible()
      })
    })
  })

  describe('Når brukeren velger uttaksalder', () => {
    it('oppdaterer valgt knapp og tegner graph', async () => {
      const user = userEvent.setup()
      const { container } = render(<Pensjonsberegning />)
      const button = await screen.findByText('68 år')
      await user.click(button)
      expect(screen.getByRole('button', { pressed: true })).toHaveTextContent(
        '68 år'
      )
      expect(
        container.getElementsByClassName('highcharts-container').length
      ).toBe(1)
    })

    it('henter ikke pensjonsavtaler når brukeren ikke har samtykket', async () => {
      const user = userEvent.setup()
      const initiateMock = vi.spyOn(
        apiSliceUtils.apiSlice.endpoints.getTpoMedlemskap,
        'initiate'
      )
      render(<Pensjonsberegning />, {
        preloadedState: { userInput: { samtykke: false } } as RootState,
      })
      const button = await screen.findByText('68 år')
      await user.click(button)
      expect(initiateMock).not.toHaveBeenCalled()
    })

    it('henter pensjonsavtaler med riktig år og måned og viser dem når brukeren har samtykket', async () => {
      const usePensjonsavtalerQueryMock = vi.spyOn(
        apiSliceUtils,
        'usePensjonsavtalerQuery'
      )
      const user = userEvent.setup()
      render(<Pensjonsberegning />, {
        preloadedState: { userInput: { samtykke: true } } as RootState,
      })
      const buttons = await screen.findAllByRole('button')
      expect(buttons).toHaveLength(12)
      await user.click(buttons[2])
      await waitFor(async () => {
        expect(await screen.findByTestId('pensjonsavtaler')).toBeInTheDocument()
      })
      expect(usePensjonsavtalerQueryMock.mock?.lastCall?.[0]).toEqual({
        antallInntektsaarEtterUttak: 0,
        uttaksperioder: [
          {
            aarligInntekt: 0,
            grad: 100,
            startAlder: 68,
            startMaaned: 1,
          },
        ],
      })
      await user.click(buttons[1])
      expect(usePensjonsavtalerQueryMock.mock?.lastCall?.[0]).toEqual({
        antallInntektsaarEtterUttak: 0,
        uttaksperioder: [
          {
            aarligInntekt: 0,
            grad: 100,
            startAlder: 67,
            startMaaned: 3,
          },
        ],
      })
    })

    it('henter pensjonsavtaler og viser riktig feilmelding ved feil', async () => {
      const user = userEvent.setup()
      mockErrorResponse('/pensjonsavtaler', {
        status: 500,
        json: "Beep boop I'm an error!",
        method: 'post',
      })
      render(<Pensjonsberegning />, {
        preloadedState: { userInput: { samtykke: true } } as RootState,
      })
      const button = await screen.findByText('68 år')
      await user.click(button)

      await waitFor(async () => {
        expect(
          await screen.findByText(
            'Vi klarte ikke å hente pensjonsavtalene dine fra Norsk Pensjon. Prøv igjen senere.'
          )
        ).toBeVisible()
      })
    })
  })
})
