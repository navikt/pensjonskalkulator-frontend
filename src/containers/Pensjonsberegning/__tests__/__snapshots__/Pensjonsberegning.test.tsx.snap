// Vitest Snapshot v1

exports[`Pensjonsberegning > viser feilmelding om henting av pensjonberegning feiler 1`] = `
<DocumentFragment>
  <div
    class="navds-alert navds-alert--error navds-alert--medium"
  >
    <svg
      aria-labelledby="title-r6"
      class="navds-alert__icon"
      fill="none"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title
        id="title-r6"
      >
        Feil
      </title>
      <path
        clip-rule="evenodd"
        d="M7.742 2.47a.75.75 0 0 1 .53-.22h7.456a.75.75 0 0 1 .53.22l5.272 5.272c.141.14.22.331.22.53v7.456a.75.75 0 0 1-.22.53l-5.272 5.272a.75.75 0 0 1-.53.22H8.272a.75.75 0 0 1-.53-.22L2.47 16.258a.75.75 0 0 1-.22-.53V8.272a.75.75 0 0 1 .22-.53L7.742 2.47Zm1.288 5.5a.75.75 0 0 0-1.06 1.06L10.94 12l-2.97 2.97a.75.75 0 1 0 1.06 1.06L12 13.06l2.97 2.97a.75.75 0 1 0 1.06-1.06L13.06 12l2.97-2.97a.75.75 0 0 0-1.06-1.06L12 10.94 9.03 7.97Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="navds-alert__wrapper navds-body-long"
    >
      <h2
        class="navds-heading navds-heading--small navds-typo--spacing"
      >
        Vi klarte ikke å hente din tidligste mulige uttaksalder. Prøv igjen senere.
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Pensjonsberegning > viser loading og deretter riktig header, tekst og knapper 1`] = `
<DocumentFragment>
  <div
    class="_container_46cd75"
  >
    <div
      class="_wrapper_f5a642"
      data-testid="tidligst-mulig-uttak"
    >
      <div
        class="_wrapperCard_f5a642"
      >
        <img
          alt=""
          class="_wrapperImage_f5a642"
          src="/pensjon/kalkulator/src/assets/piggybank.svg"
        />
        <div
          aria-live="polite"
          class="_wrapperText_f5a642"
          tabindex="-1"
        >
          <p
            class="_ingress_f5a642 _ingress__isInline_f5a642 navds-ingress"
          >
            Din opptjening i folketrygden gjør at du tidligst kan
            <br />
             ta ut alderspensjon når du er 
          </p>
          <span
            class="_highlighted_f5a642"
          >
            67 år og 1 måneder
            <div
              class="navds-help-text _helptext_f5a642"
            >
              <button
                aria-expanded="false"
                class="navds-help-text__button"
                type="button"
              >
                <svg
                  aria-labelledby="title-r2"
                  class="navds-help-text__icon"
                  fill="none"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="title-r2"
                  >
                    hjelp
                  </title>
                  <circle
                    cx="12"
                    cy="12"
                    fill="transparent"
                    r="11"
                    stroke="currentColor"
                    stroke-width="2"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M9.75 9C9.75 7.75736 10.7574 6.75 12 6.75H12.1716C13.3195 6.75 14.25 7.68054 14.25 8.82843C14.25 9.37966 14.031 9.90832 13.6412 10.2981L12.6412 11.2981C11.7504 12.1889 11.25 13.3971 11.25 14.6569C11.25 15.0711 11.5858 15.4069 12 15.4069C12.4142 15.4069 12.75 15.0711 12.75 14.6569C12.75 13.7949 13.0924 12.9682 13.7019 12.3588L14.7019 11.3588C15.373 10.6877 15.75 9.77748 15.75 8.82843C15.75 6.85212 14.1479 5.25 12.1716 5.25H12C9.92893 5.25 8.25 6.92893 8.25 9V9.5C8.25 9.91421 8.58579 10.25 9 10.25C9.41421 10.25 9.75 9.91421 9.75 9.5V9ZM12 16.5C11.4477 16.5 11 16.9477 11 17.5C11 18.0523 11.4477 18.5 12 18.5C12.5523 18.5 13 18.0523 13 17.5C13 16.9477 12.5523 16.5 12 16.5Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
                <svg
                  aria-labelledby="title-r3"
                  class="navds-help-text__icon navds-help-text__icon--filled"
                  fill="none"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="title-r3"
                  >
                    hjelp
                  </title>
                  <circle
                    cx="12"
                    cy="12"
                    fill="currentColor"
                    r="11"
                    stroke="currentColor"
                    stroke-width="2"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M9.75 9C9.75 7.75736 10.7574 6.75 12 6.75H12.1716C13.3195 6.75 14.25 7.68054 14.25 8.82843C14.25 9.37966 14.031 9.90832 13.6412 10.2981L12.6412 11.2981C11.7504 12.1889 11.25 13.3971 11.25 14.6569C11.25 15.0711 11.5858 15.4069 12 15.4069C12.4142 15.4069 12.75 15.0711 12.75 14.6569C12.75 13.7949 13.0924 12.9682 13.7019 12.3588L14.7019 11.3588C15.373 10.6877 15.75 9.77748 15.75 8.82843C15.75 6.85212 14.1479 5.25 12.1716 5.25H12C9.92893 5.25 8.25 6.92893 8.25 9V9.5C8.25 9.91421 8.58579 10.25 9 10.25C9.41421 10.25 9.75 9.91421 9.75 9.5V9ZM12 16.5C11.4477 16.5 11 16.9477 11 17.5C11 18.0523 11.4477 18.5 12 18.5C12.5523 18.5 13 18.0523 13 17.5C13 16.9477 12.5523 16.5 12 16.5Z"
                    fill="var(--a-surface-default)"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
              <div
                aria-hidden="true"
                class="navds-popover navds-help-text__popover navds-popover--hidden"
                data-placement="top"
                style="position: absolute; top: 0px; left: 0px;"
                tabindex="-1"
              >
                <div
                  class="navds-popover__content navds-body-short"
                >
                  For å starte uttak mellom 62 og 67 år må opptjeningen være høy nok. Tidspunktet er et estimat.
                </div>
                <div
                  class="navds-popover__arrow"
                  style="bottom: -0.5rem;"
                />
              </div>
            </div>
          </span>
          <p
            class="_ingress_f5a642 navds-ingress"
          >
            Jo lenger du venter, desto mer får du i året.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="_background_46cd75 _background__hasMargin_46cd75"
  >
    <div
      class="_container_46cd75"
    >
      <div
        class="_wrapper_4ca3e3"
      >
        <div
          class="_wrapperCard_4ca3e3"
        >
          <span
            class="_pin_4ca3e3"
          />
          <h2
            class="navds-heading navds-heading--xsmall"
          >
            Når vil du ta ut alderspensjon?
          </h2>
          <ul
            class="navds-chips _chipsWrapper_4ca3e3 _chipsWrapper__hasGap_4ca3e3 navds-chips--medium navds-body-short navds-body-short--small"
          >
            <li>
              <button
                aria-pressed="false"
                class="navds-chips__chip navds-chips__toggle navds-chips__toggle--action"
              >
                <span
                  class="navds-chips__chip-text"
                >
                  67 år og 1 md.
                </span>
              </button>
            </li>
            <li>
              <button
                aria-pressed="false"
                class="navds-chips__chip navds-chips__toggle navds-chips__toggle--action"
              >
                <span
                  class="navds-chips__chip-text"
                >
                  68 år
                </span>
              </button>
            </li>
            <li>
              <button
                aria-pressed="false"
                class="navds-chips__chip navds-chips__toggle navds-chips__toggle--action"
              >
                <span
                  class="navds-chips__chip-text"
                >
                  69 år
                </span>
              </button>
            </li>
            <li>
              <button
                aria-pressed="false"
                class="navds-chips__chip navds-chips__toggle navds-chips__toggle--action"
              >
                <span
                  class="navds-chips__chip-text"
                >
                  70 år
                </span>
              </button>
            </li>
            <li>
              <button
                aria-pressed="false"
                class="navds-chips__chip navds-chips__toggle navds-chips__toggle--action"
              >
                <span
                  class="navds-chips__chip-text"
                >
                  71 år
                </span>
              </button>
            </li>
            <li>
              <button
                aria-pressed="false"
                class="navds-chips__chip navds-chips__toggle navds-chips__toggle--action"
              >
                <span
                  class="navds-chips__chip-text"
                >
                  72 år
                </span>
              </button>
            </li>
            <li>
              <button
                aria-pressed="false"
                class="navds-chips__chip navds-chips__toggle navds-chips__toggle--action"
              >
                <span
                  class="navds-chips__chip-text"
                >
                  73 år
                </span>
              </button>
            </li>
            <li>
              <button
                aria-pressed="false"
                class="navds-chips__chip navds-chips__toggle navds-chips__toggle--action"
              >
                <span
                  class="navds-chips__chip-text"
                >
                  74 år
                </span>
              </button>
            </li>
            <li>
              <button
                aria-pressed="false"
                class="navds-chips__chip navds-chips__toggle navds-chips__toggle--action"
              >
                <span
                  class="navds-chips__chip-text"
                >
                  75 år
                </span>
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    class="_background_46cd75 _background__lightblue_46cd75"
  >
    <div
      class="_container_46cd75"
    >
      <section
        class="_section_095025"
      >
        <div
          class="_innerwrapper_095025"
        >
          <button
            class="_button_095025 navds-button navds-button--secondary navds-button--medium"
          >
            <span
              class="navds-label"
            >
              Tilbake til start
            </span>
          </button>
          <button
            class="_button_095025 _button__avbryt_095025 navds-button navds-button--tertiary navds-button--medium"
          >
            <span
              class="navds-label"
            >
              Avbryt
            </span>
          </button>
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
